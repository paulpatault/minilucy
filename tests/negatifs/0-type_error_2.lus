type t = A | B | C

node main0 (x : t) returns (o: bool);
let
  o = merge x
        (A't -> true  when A(x))
        (B't -> 1     when B(x))
        (C't -> false when C(x));
tel
