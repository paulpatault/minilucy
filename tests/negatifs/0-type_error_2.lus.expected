
/**************************************/
/* Sugar automaton print              */
/**************************************/
type inductive_bool =
  | False
  | True
type t =
  | A
  | B
  | C

node main0(x : adt(t)) returns (o : bool)
var ;
let
  (o) = merge x 
          (t(A) -> true when A(x))
          (t(B) -> 1 when B(x))
          (t(C) -> false when C(x));
tel

[31m[1mTyping error: [0mthis expression has type bool but is expected to have type int

